sceptre/stack.py

```diff
@@ -157,9 +157,37 @@ def template(self):
                 self.config["template_path"],
             )
 
+            s3_details = None
+            if "template_bucket_name" in self.environment_config:
+                template_key = "/".join([
+                    self.region, self._environment_path,
+                    "{stack_name}-{time_stamp}.json".format(
+                        stack_name=self.external_name,
+                        time_stamp=datetime.datetime.utcnow().strftime(
+                            "%Y-%m-%d-%H-%M-%S-%fZ"
+                        )
+                    )
+                ])
+
+                if "template_key_prefix" in self.environment_config:
+                    prefix = self.environment_config["template_key_prefix"]
+                    template_key = "/".join([
+                        prefix.strip("/"), template_key
+                    ])
+
+                s3_details = {
+                     "bucket_name": self.environment_config[
+                        "template_bucket_name"
+                     ],
+                     "bucket_key": template_key,
+                     "region": self.region
+                }
+
```

Comment:
Could this be delegated to a separate function? It looks a bit long to me and it seems like the only part you really need is the s3_details, i.e., you don't do any weird things with manipulating self here or anything.